1-mandatory/2-people-i-know.js

```diff
@@ -393,15 +397,43 @@ You can see who people's colleagues are by seeing the "colleagues" array in each
 This time, I only want the full names ("<firstname> <lastname>") of my friends who are colleagues of hers.
 */
 
-let friendsWhoAreColleaguesOfStacie = [];
+let friendsWhoAreColleaguesOfStacie = []
+function Villarreal (arrayOfFriends) {
+  const result = [];
+  for (let i = 0; i < arrayOfFriends.length; i++) {
+    for (let j = 0; j < arrayOfFriends[i].colleagues.length; j++) {
+      if (arrayOfFriends[i].colleagues[j].name == "Stacie Villarreal") {
+        result.push(
+          `${arrayOfFriends[i].name.first} ${arrayOfFriends[i].name.last}`
+        );
+      }
+    }
+  }
```

Comment:
This works fine, but I think the readability of the code can be improved with js native array methods instead of using nested loops. 

e.g. 
with forEach

```
arr.forEach((friend) => {
    const colleagueWithStacie = friend.colleagues.find(
      (colleague) => colleague.name === "Stacie Villarreal"
    );

    if (colleagueWithStacie) {
      result.push(`${friend.name.first} ${friend.name.last}`);
    }
  });
```

with reduce: 

```
return arr.reduce((acc, friend) => {
    const colleagueWithStacie = friend.colleagues.find(
      (colleague) => colleague.name === "Stacie Villarreal"
    );

    if (colleagueWithStacie) {
      acc.push(`${friend.name.first} ${friend.name.last}`);
    }

    return acc;
  }, []);
```